
// // --------------------------- Promises ---------------------------//

// const promise = new Promise((resolve, reject) => {
//     setTimeout(() => {resolve('resolve is called I/O operation successful')}, 3000);
// });

// const generalFunc = () => {
//     console.log("Inside general function");
//     promise.then(
//         parameters => {
//             console.log(parameters);
//             return ("Hello from first .then")
//         }

//     )
//     .then( (parametersFromFirst) => {
//         console.log(parametersFromFirst)
//     })
//     .catch(
//         error => console.log(error)
//     )
// }

// generalFunc();
